docs/docs/055-platforms/awscdk.md

```diff
@@ -6,50 +6,161 @@ description: AWS CDK platform
 keywords: [Wing reference, Wing language, language, Wing language spec, Wing programming language, aws, awscdk, amazon web services, cloudformation]
 ---
 
-The `@winglang/platform-awscdk` [platform](../02-concepts/03-platforms.md) compiles your program for the AWS CDK (CloudFormation).
+The `@winglang/platform-awscdk` [platform](../02-concepts/03-platforms.md) compiles your program for
+the AWS CDK and deployed through the CDK CLI (and AWS CloudFormation).
+
+## Prerequisites
+
+* Install the AWS CDK (or use via `npx cdk`):
+  ```sh
+  npm i aws-cdk
+  ```
+
+* Install the Wing `awscdk` platform:
+  ```sh
+  npm i @winglang/platform-awscdk
+  ```
 
 ## Usage
 
-You will need to install the `@winglang/platform-awscdk` library in order to use this platform.
+Let's create `main.w` with our Wing program:
+
+```js
+bring cloud;
+
+new cloud.Bucket();
+```
+
+> At this point, you can just run `wing it` (or `wing run`) to open the Wing Simulator.
+
+To use Wing with the AWS CDK, we will need to tell the CDK CLI to run `wing compile` as the CDK app
+and that the synthesis output is in the Wing's target directory.
+
+This can be done by creating a `cdk.json` file manually or through `cdk init` and editing the `app`
+and the `output` fields:
+
+```json
+{
+  "app": "CDK_STACK_NAME='MyStack' CDK_AWS_ACCOUNT='111111555555' CDK_AWS_REGION='us-east-1' wing compile --platform @winglang/platform-awscdk main.w",
+  "output": "target/main.awscdk",
+
+  // ... rest of cdk.json
+}
+```
+
+The `awscdk` platform uses the following environment variables as configuration options:
+
+* `CDK_STACK_NAME` (required) - sets the CloudFormation stack name to use.
+- `CDK_AWS_ACCOUNT` and `CDK_AWS_REGION` (optional) - the AWS environment for deployment and context
+  lookups (e.g. VPC lookups). The default is to use the AWS account region defined in the CLI
+  environment.
+
+Now, the AWS CDK CLI will work as normal:
+
+* `npx cdk bootstrap` bootstrap your AWS account for AWS CDK use (once per account/region).
+* `npx cdk deploy` deploy the Wing app to your default AWS account/region.
+* `npx cdk synth` synthesize output to `cdk.out`.
```

Comment:
Shouldn't the platform be responsible for this? The workflow described in this document no longer feels like a wing application, but rather an aws cdk one that happens to be written in wing.
Does the CLI expose anything programmatically we can use to solve this instead?

Reply:
Wing doesn't take care of bootstrapping/deployment/diffs. Similarly to how in Terraform we use the Terraform CLI for this.

It is an AWS CDK application written in Wing for all intents and purposes.

The CDK CLI doesn't have a good programmatic API.

I believe this DX is more idiomatic for CDK users, which is the purpose of this platform basically.

Reply:
We should provide something like `wing new awscdk` to make this seamless, but for now, I am fine with requiring people to edit `cdk.json`.

Reply:
My comment was specifically marked on `synth` i.e. `compile`, where wing does take ownership. This PR would take `wing` out of that loop that we claim to be a part of. Obviously it's still involved, but it's no longer the "entrypoint".

Eventually, I think wing should take ownership of common things like `diff` and `deploy` too, but until then I think it benefits us to not lostt

Reply:
Im not in love with this flow, but pragmatically I understand why this is the direction to go for now. 

I dont want to get too crazy with platform hooks but in theory we could introduce a `compile()` hook to platforms, and the `awscdk` platform might be able to obfuscate all this from the users who would still just run `wing compile`

But thats just a "from the hip" thought.

>The CDK CLI doesn't have a good programmatic API.
^😢

Reply:
You can still do `wing compile` and it will behave like today. But when you want to deploy, you just deploy with `cdk deploy`.

The only difference from the current experience is that instead of doing `cdk deploy --app target/main.awscdk` you just run `cdk deploy`. It's way more idiomatic and much nicer from every aspect as far as I am concerned.

Reply:
> You can still do wing compile and it will behave like today. But when you want to deploy, you just deploy with cdk deploy.

Sure, but `wing compile` won't work correctly so it has effectively been removed as the correct option. Even though it just affects context lookup, we're still basically saying "don't use `wing compile`"

Reply:
This is true for all AWS CDK languages. If you just compile + synthesize a CDK program without going through the CDK CLI context providers won't work.

Reply:
Not sure what's disappointing about it TBH, but ok.

Reply:
Not sure what's disappointing about it TBH, but ok.